code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -26,6 +26,8 @@
 	del_on_death = TRUE
 	deathmessage = "collapses in a shattered heap."
 	var/construct_type = "shade"
+	/// The body/brain of the player inside this construct, transferred over from the soulstone.
+	var/mob/living/held_body
```

Comment:
this needs to be properly GC'd

Reply:
Read comment above

Reply:
Same thing here, also needs to be nulled in Destroy.